From be34caa9498a41766cdedf43613edf2e9600d896 Mon Sep 17 00:00:00 2001
From: Max Reitz <mreitz@redhat.com>
Date: Sat, 13 Jun 2015 16:22:13 +0200
Subject: [PATCH 19/42] qcow2: Use sizeof(**refcount_table)

Message-id: <1434212556-3927-20-git-send-email-mreitz@redhat.com>
Patchwork-id: 66038
O-Subject: [RHEL-7.2 qemu-kvm PATCH 19/42] qcow2: Use sizeof(**refcount_table)
Bugzilla: 1129893
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Fam Zheng <famz@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

BZ: 1129893

When implementing variable refcounts, we want to be able to easily find
all the places in qemu which are tied to a certain refcount order.
Replace sizeof(uint16_t) in the check code by sizeof(**refcount_table)
so we can later find it more easily.

Signed-off-by: Max Reitz <mreitz@redhat.com>
Reviewed-by: Eric Blake <eblake@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit 78fb328e854542d79bebe54f3a426cba6d46dbf1)

Signed-off-by: Max Reitz <mreitz@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 block/qcow2-refcount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/qcow2-refcount.c b/block/qcow2-refcount.c
index 08b2b62..cb78f05 100644
--- a/block/qcow2-refcount.c
+++ b/block/qcow2-refcount.c
@@ -1613,7 +1613,7 @@ static int check_refblocks(BlockDriverState *bs, BdrvCheckResult *res,
                                                   *nb_clusters);
                         memset(&(*refcount_table)[old_nb_clusters], 0,
                                (*nb_clusters - old_nb_clusters) *
-                               sizeof(uint16_t));
+                               sizeof(**refcount_table));
                     }
                     (*refcount_table)[cluster]--;
                     inc_refcounts(bs, res, *refcount_table, *nb_clusters,
-- 
1.8.3.1

